drivers/sensor/st/stm32_vbat/stm32_vbat.c

```diff
@@ -101,6 +101,11 @@ static int stm32_vbat_init(const struct device *dev)
 
 	k_mutex_init(&data->mutex);
 
+	if (data->adc == NULL) {
```

Comment:
This shouldn't be needed as the "device_is_ready" function allows NULL to be passed, but I see there is a pointer dereference that could be a problem.

Reply:
I can do that but then there will be no difference between 'adc driver init failure' vs 'adc node not enabled in devicetree'

cc @erwango, any preference?

Reply:
Is it possible if `DT_HAS_ST_STM32_ADC_ENABLED` has been added?

Reply:
As mentioned in the commit,  that checks if we have at least an adc enabled in the DT, but not necessarily the one used by vbat

Reply:
> DT_HAS_ST_STM32_ADC_ENABLED

Yes, another adc instance could be enabled, but not the one we need.

Reply:
> As mentioned in the commit, that checks if we have at least an adc enabled in the DT, but not necessarily the one used by vbat

Fair enough.

Reply:
> I can do that but then there will be no difference between 'adc driver init failure' vs 'adc node not enabled in devicetree'

Indeed, `'adc node not enabled in devicetree'` can be a tricky one, it's better if we're able to provide a clear error message on this specific issue.